src/main/java/net/minecraftforge/network/HandshakeMessages.java

```diff
@@ -99,6 +117,16 @@ public void encode(FriendlyByteBuf output)
 
             output.writeVarInt(registries.size());
             registries.forEach(output::writeResourceLocation);
+            
+            // The list of synced datapack registry names is not sent in 1.18.2 if the list is empty.
+            // TODO 1.19: should send an empty list if the list is empty.
+            Set<ResourceKey<? extends Registry<?>>> dataPackRegistries = DataPackRegistriesHooks.getSyncedCustomRegistries();
+            int size = dataPackRegistries.size();
+            if (size > 0)
+            {
+                output.writeVarInt(size);
+                dataPackRegistries.forEach(key -> output.writeResourceLocation(key.location()));
```

Comment:
Could you instead use `FriendlyByteBuf#writeCollection` here (which is the counterpart to `FriendlyByteBuf#readCollection`)?

Reply:
Changed in next push